File: JAVA/ml/ML_Lists.java
package ml;

import java.util.Map;

import org.json.JSONObject;

import base.ML_Rest;

/**
 * Managing Lists Class for MailerSoft.
 * 
 * @author Vaidotas Strazdas
 *
 */
public class ML_Lists extends ML_Rest {

	/**
	 * Constructor.
	 * 
	 * @param apiKey API Key
	 */
    public ML_Lists(String apiKey) {
		super(apiKey);
		this.name = "lists";
		this.setPath(null);
    }
	
    /**
     * setId() overloaded, so the programmer won't need any casting.
     */
	public ML_Lists setId(long id) {
		super.setId(id);
		return this;
	}

	/**
	 * Get active on the list.
	 * 
	 * @param data Specific data about the list.
	 * @return Active users on the list.
	 * @throws Exception
	 * @see JSONObject
	 */
    public JSONObject getActive(Map<String, Object> data) throws Exception {
    	JSONObject result = this.get(data, "active");
        return result;
    }

    /**
     * Overloaded getActive() method.
     * 
     * @return Active users on the list.
     * @throws Exception
     * @see JSONObject
     */
    public JSONObject getActive() throws Exception {
    	JSONObject result = this.getActive(null);
        return result;
    }

    /**
     * Get unsubscribed users.
     * 
     * @param data Specific data about the list.
     * @return Unsubscribed users.
     * @throws Exception
     * @see JSONObject
     */
    public JSONObject getUnsubscribed(Map<String, Object> data) throws Exception {
    	JSONObject result = this.get(data, "unsubscribed");
        return result;
    }

    /**
     * Overloaded getUnsubscribed() method.
     * 
     * @return Unsubscribed users.
     * @throws Exception
     * @see JSONObject
     */
    public JSONObject getUnsubscribed() throws Exception {
    	JSONObject result = this.getUnsubscribed(null);
        return result;
    }

    /**
     * Get bounced users.
     * 
     * @param data Specific data about the list.
     * @return Bounced users.
     * @throws Exception
     * @see JSONObject
     */
    public JSONObject getBounced(Map<String, Object> data) throws Exception {
    	JSONObject result = this.get(data, "bounced");
        return result;
    }

    /**
     * Overloaded getBounced() method.
     * 
     * @return Bounced users.
     * @throws Exception
     * @see JSONObject
     */
    public JSONObject getBounced() throws Exception {
    	JSONObject result = this.getBounced(null);
        return result;
    }
	
}

File: Python/ML_Lists.py
from ML_Rest import ML_Rest

class ML_Lists(ML_Rest):
    """
    ML_Lists class is intended to be used for getting information about the lists of recipients.
    """
    
    def __init__(self, api_key):
        """
        Constructor for the ML_Lists class.
        
        :param api_key: Your API Key for MailerSoft.
        :type api_key: str
        """
        self._mc_parent = ML_Rest
        self._mc_parent.__init__(self, api_key)
        self._name = "lists"
        self._set_path()
    
    def get_active(self, data={}):
        """
        Get all active subscribers in given list.
        You do not call this method before calling set_id().
        
        :param data:
            You can set limits of results in one page (default is 1000).
            You also can navigate through records by increasing 'page'.
            So, you may use data = {'results': 100, 'page': 1} to get 100 results in the 2nd (0 is the 1st) page.
        :type data: dict

        :returns: JSONObject with subscribers in given list.
        :rtype: json
        
        """
        return self._mc_parent.get(self, data, "active")
    
    def get_unsubscribed(self, data={}):
        """
        Get all unsubscribed subscribers in given list.
        You do not call this method before calling set_id().
        
        :param data:
            You can set limits of results in one page (default is 1000).
            You also can navigate through records by increasing 'page'.
            So, you may use data = {'results': 100, 'page': 1} to get 100 results in the 2nd (0 is the 1st) page.
        :type data: dict

        :returns: JSONObject with unsubscribed subscribers in given list.
        :rtype: json
        
        """
        return self._mc_parent.get(self, data, "unsubscribed")
    
    def get_bounced(self, data={}):
        """
        Get all bounced subscribers in given list.
        You do not call this method before calling set_id().
        
        :param data:
            You can set limits of results in one page (default is 1000).
            You also can navigate through records by increasing 'page'.
            So, you may use data = {'results': 100, 'page': 1} to get 100 results in the 2nd (0 is the 1st) page.
        :type data: dict

        :returns: JSONObject with bounced subscribers in given list.
        :rtype: json
        
        """
        return self._mc_parent.get(self, data, "bounced")